File: generate_random_data_sparse high_variance.r
#!/usr/bin/env Rscript

p <- 7
q <- 2
result <- c()
for ( i in 0:(p-1) ){
	for ( i2 in 0:(q^i-1) ){
		r <- sample(0:( q^(p-i)-1 ),1)
		result <- c( result, r )
	}
}
result <- sample( result )
print( result )

File: generate_random_data_sparse high_variance.py
#!/usr/bin/env python3

import random

n = 7
result = []
for i in range( n ):
	for i2 in range( 2**i ):
		result.append( random.randint( 0, 2**(n-i) - 1 ))
random.shuffle( result )

print( result )
